### PR TITLE
HTTP Server Spec Acceptance Tests

### PR DESCRIPTION
**Pull request by evan-m-jackson**
- Refactored code to respond to different url paths in the HTTP Server Spec Acceptance Tests
- Refactored port to 5000
- Refactored file names and separated project files by request and response
- Refactored tests into separate files for Integration, Request and Response Tests
- Added in `web/` file content from `HTTP-Server-Spec`

### CODE REVIEW

File: HTTPServerProject/HTTPServerResponse/ResponseByPath.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.IO;
+using HTTPServerProject.Responses;
+using HTTPServerProject.Responses.Write;
+using HTTPServerProject.WriteStream;
+
+namespace HTTPServerProject.Path
```
Comment (paytonrules) [inline HTTPServerProject/HTTPServerResponse/ResponseByPath.cs:10]: Why aren't there any unit tests for this code?

**Comment by paytonrules** [comment]
*  Change Requested: Can you get the version of HttpSpec you're using into this repository with directions on how to run them. It makes it a lot easier to track your progress.

You'll want to use a Git subtree https://www.atlassian.com/git/tutorials/git-subtree to include a separate repo.

File: HTTPServerProject/HTTPServerResponse/ResponseByPath.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+{
+    public class RequestPath
+    {
+        IWriteStreams writer;
+
+        public Dictionary<string, object> pathDict = new Dictionary<string, object>();
```
Comment (paytonrules) [inline HTTPServerProject/HTTPServerResponse/ResponseByPath.cs]: Gonna highlight this here because it looks like you started down a path with `pathDict`. 

So I mentioned changing this code doesn't cause any unit tests to fail, which is a problem. Remember the 1st rule of TDD - no code without a failing unit test. 

You'll find your unit tests, at least at the start, are redundant. But the feature tests you're making pass are higher on the testing triangle, somewhere between acceptance and integration. They typically only cover positive cases, and they require the entire server running. They also don't provide much _design_ pressure, and that's where I think unit tests would help.

If you write unit tests you'll find yourself realizing are both redundant and repetitive - like the code is here, and that's a hint. You're tests are telling you "hey your code could be more generic, then you'd need less tests AND less code." I think you started to see this and that's why the pathDict is here - if every path was just data in a lookup table, then your unit tests would verify that the looking up the response from the lookup table worked, and your feature tests would validate the _specific_ cases like "method_options".  

If you move in that direction you'll probably run into a weird corner case or two but ultimately end up with code that hits both technical requirements.

**Comment by paytonrules** [comment]
A note on commit messages: They should have a short summary on the top (less than 50 characters). Most tools - including GitHub - are modeled around this limitation. It's kinda silly but not too hard to stick to.

See here for pretty good guidelines: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

**Review by paytonrules — COMMENTED** [review]
Can you get the `.idea` directory out of the repo as well - via the .gitignore. That's going to have a bunch of editor specific stuff in it.
